--- 
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
bibliography: [book.bib]
biblio-style: apalike
link-citations: yes
github-repo: rstudio/bookdown-demo
description: ""
---
# Our dataset {-}

18 water samples collected from rivers and lakes near oil-sands extraction sites in the province of Alberta, Canada. Collected water was filtered and DNA collected on the filers was extracted and sequenced. For each sample collected, two different marker genes were used to amplify and sequence distinct DNA regions : the **12S** ribosomal RNA gene and the mitochondrial cytochrome oxidase subunit (**COI**). 

```{r, echo = FALSE, warning = FALSE}
library(DT)
primer_info_table = read.csv(file = "data/primer_table.csv", check.names = FALSE)
datatable(primer_info_table, rownames = FALSE, caption="Amplified regions details", options=list(scrollX=T, dom = 't', paging=FALSE, ordering=FALSE,
                                                            initComplete = JS("function(settings, json) {",
                                                                              "$(this.api().table().header()).css({'font-size': '10px'});",
                                                                              "}"))) %>% formatStyle(columns = colnames(.$x$data), `font-size` = '10px')
```

## Alberta fishes {-}

[Fishbase](https://www.fishbase.se/search.php) was used to generate the following list of Alberta freshwater fishes. 

```{r, echo = FALSE, warning = FALSE}
library(DT)
fishbase_alberta = read.csv(file = "data/fishbase_alberta_freswater.csv", check.names = FALSE)
datatable(fishbase_alberta, rownames = FALSE, options=list(initComplete = JS("function(settings, json) {",
                                                                              "$(this.api().table().header()).css({'font-size': '11px'});",
                                                                              "}"))) %>% formatStyle(columns = colnames(.$x$data), `font-size` = '11px')
```

## Worklfows and their associated available databases {-}

```{r, echo = FALSE, warning = FALSE}
library(DT)
workflow_database = read.csv(file = "data/workflow_databases.csv", check.names = FALSE)
datatable(workflow_database, rownames = FALSE, options=list(scrollX=T, dom = 't', paging=FALSE, ordering=FALSE,
                                                            initComplete = JS("function(settings, json) {",
                                                                              "$(this.api().table().header()).css({'font-size': '11px'});",
                                                                              "}"))) %>% formatStyle(columns = colnames(.$x$data), `font-size` = '11px')
```

**Links** 

- [1] [Barcode of life (BOLD)](https://www.boldsystems.org/)
- [2] [barque workflow](https://github.com/enormandeau/barque) 
- [3] [COI formatted for Barque](https://www.ibis.ulaval.ca/services/bioinformatique/barque_databases/)
- [4] [DADA2 workflow](https://benjjneb.github.io/dada2/ITS_workflow.html)
- [5] [Eukaryote CO1 Classifier](https://github.com/terrimporter/CO1Classifier/releases/tag/RDP-COI-v5.1.0) 
- [6] [Mitochondrial Genome Database of Fish (MitoFish)](https://mitofish.aori.u-tokyo.ac.jp/) 
- [7] [National Center for Biotechnology Information (NCBI)](https://www.ncbi.nlm.nih.gov/)
- [8] [Ribosomal Database Project (RDP) classifier](https://github.com/rdpstaff/classifier)
- [9] [VSEARCH](https://github.com/torognes/vsearch)
- [10] [12S fish Classifier v1.0.1](https://github.com/terrimporter/12SfishClassifier/releases/tag/v1.0.1)
- [11] [12S formatted for barque](https://github.com/enormandeau/barque/blob/master/03_databases/12S.fasta.gz)

### Representation of Alberta freshwater fishes in databases {-}

Original code to evaluate if the fish from Alberta were present in each of the databases ([3](https://www.ibis.ulaval.ca/services/bioinformatique/barque_databases/), [5](https://github.com/terrimporter/CO1Classifier/releases/tag/RDP-COI-v5.1.0) , [10](https://github.com/terrimporter/12SfishClassifier/releases/tag/v1.0.1), [11](https://github.com/enormandeau/barque/blob/master/03_databases/12S.fasta.gz)) can be found [here][Evaluation representation in databases]. 

```{r, echo = FALSE, message = FALSE, warning = FALSE}
#library(shiny)
library(DT)
library(dplyr)

aff_databases = read.csv(file = "data/represensation_albertafish_databases.csv", check.names = FALSE, row.names = 1)
datatable(aff_databases, rownames = FALSE, options = list(scrollX = T, dom = 't', paging=FALSE, ordering=FALSE,
                                                        initComplete = JS("function(settings, json) {",
                                                                              "$(this.api().table().header()).css({'font-size': '11px'});",
                                                                              "}"))) %>% formatStyle(columns = colnames(.$x$data), `font-size` = '11px') %>% 
  formatStyle(names(aff_databases), 
              backgroundColor = styleEqual(c("Yes","No"), c("#DeF7E9", "#ff7f7f")), 
              backgroundSize = "50%", 
              backgroundPosition = 'center')
```

## Comparing pipelines with available databases {-}

### DADA2 {-}

#### 12S {-}

#### COI {-}

839 ASV | from these 839 ASVs, 20 (2.38 %) are species with a bootstrap level above 70 % | from these 20 species, 8 (40 %) are from class actinopteri | from these 8 actinopteri, 7 (87.5 %) are species of freshwater fishes found in Alberta. 

### Barque {-}

## Curating databases {-}

https://forum.qiime2.org/t/building-a-coi-database-from-bold-references/16129

Using `seqkit` tool to filter databases in order to keep only sequences of freshwater fishes found in Alberta 

```{bash, eval = FALSE}
seqkit grep -v -n -f id_list.txt in.fasta > out.fasta
```
