<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Supplementary workflow | Fish eDNA</title>
  <meta name="description" content="Supplementary workflow | Fish eDNA" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="Supplementary workflow | Fish eDNA" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Supplementary workflow | Fish eDNA" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="taxonomy-cross-validation-by-identity-taxxi.html"/>
<link rel="next" href="supplementary-data.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.33/datatables.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#objectives"><i class="fa fa-check"></i>Objectives</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#dataset"><i class="fa fa-check"></i>Dataset</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#alberta-fishes"><i class="fa fa-check"></i>Alberta fishes</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#methods-and-their-associated-available-databases"><i class="fa fa-check"></i>Methods and their associated available databases</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#representation-of-alberta-freshwater-fishes-in-databases"><i class="fa fa-check"></i>Representation of Alberta freshwater fishes in databases</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="required-files-scripts-and-programs.html"><a href="required-files-scripts-and-programs.html"><i class="fa fa-check"></i>Required files, scripts and programs</a>
<ul>
<li class="chapter" data-level="" data-path="required-files-scripts-and-programs.html"><a href="required-files-scripts-and-programs.html#programs"><i class="fa fa-check"></i>Programs</a></li>
<li class="chapter" data-level="" data-path="required-files-scripts-and-programs.html"><a href="required-files-scripts-and-programs.html#databases"><i class="fa fa-check"></i>Databases</a></li>
<li class="chapter" data-level="" data-path="required-files-scripts-and-programs.html"><a href="required-files-scripts-and-programs.html#scripts"><i class="fa fa-check"></i>Scripts</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="taxonomy-cross-validation-by-identity-taxxi.html"><a href="taxonomy-cross-validation-by-identity-taxxi.html"><i class="fa fa-check"></i>Taxonomy cross validation by identity (TAXXI)</a>
<ul>
<li class="chapter" data-level="" data-path="taxonomy-cross-validation-by-identity-taxxi.html"><a href="taxonomy-cross-validation-by-identity-taxxi.html#description"><i class="fa fa-check"></i>Description</a></li>
<li class="chapter" data-level="" data-path="taxonomy-cross-validation-by-identity-taxxi.html"><a href="taxonomy-cross-validation-by-identity-taxxi.html#methods"><i class="fa fa-check"></i>Methods</a></li>
<li class="chapter" data-level="" data-path="taxonomy-cross-validation-by-identity-taxxi.html"><a href="taxonomy-cross-validation-by-identity-taxxi.html#implementing-cvi-with-our-data"><i class="fa fa-check"></i>Implementing CVI with our data</a></li>
<li class="chapter" data-level="" data-path="taxonomy-cross-validation-by-identity-taxxi.html"><a href="taxonomy-cross-validation-by-identity-taxxi.html#generate-benchmark-datasets"><i class="fa fa-check"></i>Generate benchmark datasets</a></li>
<li class="chapter" data-level="" data-path="taxonomy-cross-validation-by-identity-taxxi.html"><a href="taxonomy-cross-validation-by-identity-taxxi.html#generate-predictions"><i class="fa fa-check"></i>Generate predictions</a>
<ul>
<li class="chapter" data-level="" data-path="taxonomy-cross-validation-by-identity-taxxi.html"><a href="taxonomy-cross-validation-by-identity-taxxi.html#rdp-classifier"><i class="fa fa-check"></i>RDP Classifier</a></li>
<li class="chapter" data-level="" data-path="taxonomy-cross-validation-by-identity-taxxi.html"><a href="taxonomy-cross-validation-by-identity-taxxi.html#vsearch"><i class="fa fa-check"></i>VSEARCH</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="taxonomy-cross-validation-by-identity-taxxi.html"><a href="taxonomy-cross-validation-by-identity-taxxi.html#cvi-metrics"><i class="fa fa-check"></i>CVI metrics</a>
<ul>
<li class="chapter" data-level="" data-path="taxonomy-cross-validation-by-identity-taxxi.html"><a href="taxonomy-cross-validation-by-identity-taxxi.html#script-to-evaluate-prediction"><i class="fa fa-check"></i>Script to evaluate prediction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="supplementary-workflow.html"><a href="supplementary-workflow.html"><i class="fa fa-check"></i>Supplementary workflow</a>
<ul>
<li class="chapter" data-level="" data-path="supplementary-workflow.html"><a href="supplementary-workflow.html#curating-databases-for-distance-matrix"><i class="fa fa-check"></i>Curating databases for distance matrix</a>
<ul>
<li class="chapter" data-level="" data-path="supplementary-workflow.html"><a href="supplementary-workflow.html#rdp"><i class="fa fa-check"></i>RDP</a></li>
<li class="chapter" data-level="" data-path="supplementary-workflow.html"><a href="supplementary-workflow.html#barque"><i class="fa fa-check"></i>Barque</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supplementary-workflow.html"><a href="supplementary-workflow.html#worklow-used"><i class="fa fa-check"></i>Worklow used</a>
<ul>
<li class="chapter" data-level="" data-path="supplementary-workflow.html"><a href="supplementary-workflow.html#dada2"><i class="fa fa-check"></i>DADA2</a></li>
<li class="chapter" data-level="" data-path="supplementary-workflow.html"><a href="supplementary-workflow.html#barque-1"><i class="fa fa-check"></i>Barque</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supplementary-workflow.html"><a href="supplementary-workflow.html#curating-databases-for-alberta-freshwater-fish"><i class="fa fa-check"></i>Curating databases for Alberta freshwater fish</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supplementary-data.html"><a href="supplementary-data.html"><i class="fa fa-check"></i>Supplementary data</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Fish eDNA</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="supplementary-workflow" class="section level1 unnumbered hasAnchor">
<h1>Supplementary workflow<a href="supplementary-workflow.html#supplementary-workflow" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="curating-databases-for-distance-matrix" class="section level2 unnumbered hasAnchor">
<h2>Curating databases for distance matrix<a href="supplementary-workflow.html#curating-databases-for-distance-matrix" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Required files :</p>
<ul>
<li>barque COI <a href="https://www.ibis.ulaval.ca/services/bioinformatique/barque_databases/">3</a> (<code>barque_coi.fasta</code>)</li>
<li>RDP COI V5.1.0 <a href="https://github.com/terrimporter/CO1Classifier/releases/tag/RDP-COI-v5.1.0">5</a> (<code>rdp_coi.fasta</code>)</li>
</ul>
<div id="rdp" class="section level3 unnumbered hasAnchor">
<h3>RDP<a href="supplementary-workflow.html#rdp" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Filter out sequence which do not belong to class Actinopteri</li>
</ol>
<div class="sourceCode" id="cb19"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb19-1"><a href="supplementary-workflow.html#cb19-1" tabindex="-1"></a><span class="ex">seqkit</span> grep <span class="at">-r</span> <span class="at">-n</span> <span class="at">-p</span> <span class="st">&#39;.*Actinopteri*&#39;</span> rdp_coi.fasta <span class="at">-o</span> chord_rdp_coi.fasta</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Extract sequences header and print into txt file</li>
</ol>
<div class="sourceCode" id="cb20"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb20-1"><a href="supplementary-workflow.html#cb20-1" tabindex="-1"></a><span class="fu">grep</span> <span class="st">&quot;&gt;&quot;</span> chord_rdp_coi.fasta <span class="kw">|</span> <span class="fu">sed</span> <span class="at">-e</span> <span class="st">&quot;s/&gt;//&quot;</span> <span class="op">&gt;</span> rdp_coi_header.txt</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>The file with the sequence headers can then be passed to R to generate a list of which headers to keep. As per methodology by Edgar (2018), because the databases have a highly uneven numbers of sequences per genus, a subset is create by imposing a maximum of 10 sequences per genus. Headers are sampled at random to meet this constraint.</li>
</ol>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="supplementary-workflow.html#cb21-1" tabindex="-1"></a><span class="do">#### ------------------------- Load libraries ------------------------------------------------------------------####  </span></span>
<span id="cb21-2"><a href="supplementary-workflow.html#cb21-2" tabindex="-1"></a></span>
<span id="cb21-3"><a href="supplementary-workflow.html#cb21-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb21-4"><a href="supplementary-workflow.html#cb21-4" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb21-5"><a href="supplementary-workflow.html#cb21-5" tabindex="-1"></a></span>
<span id="cb21-6"><a href="supplementary-workflow.html#cb21-6" tabindex="-1"></a><span class="do">#### ------------------------- Load databases sequence header -------------------------------------------------- #### </span></span>
<span id="cb21-7"><a href="supplementary-workflow.html#cb21-7" tabindex="-1"></a></span>
<span id="cb21-8"><a href="supplementary-workflow.html#cb21-8" tabindex="-1"></a>rdpcoi <span class="ot">=</span> <span class="fu">read.table</span>(<span class="st">&quot;/home/kvilleneuve/fish_edna/database/taxxi_cvi/COI_DB/rdp_coi_header.txt&quot;</span>,<span class="at">sep =</span> <span class="st">&quot;;&quot;</span>)</span>
<span id="cb21-9"><a href="supplementary-workflow.html#cb21-9" tabindex="-1"></a></span>
<span id="cb21-10"><a href="supplementary-workflow.html#cb21-10" tabindex="-1"></a><span class="do">#### ------------------------- Format dataframes -------------------------------------------------------------- #### </span></span>
<span id="cb21-11"><a href="supplementary-workflow.html#cb21-11" tabindex="-1"></a><span class="fu">names</span>(rdpcoi) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;ID&quot;</span>, <span class="st">&quot;Cellular_organism&quot;</span>,<span class="st">&quot;Superkingdom&quot;</span>, <span class="st">&quot;Kingdom&quot;</span>, <span class="st">&quot;Phylum&quot;</span>, <span class="st">&quot;Class&quot;</span>, <span class="st">&quot;Order&quot;</span>, <span class="st">&quot;Family&quot;</span>, <span class="st">&quot;Genus&quot;</span>, <span class="st">&quot;Species&quot;</span>)</span>
<span id="cb21-12"><a href="supplementary-workflow.html#cb21-12" tabindex="-1"></a>rdp_coi_actino <span class="ot">=</span> <span class="fu">subset</span>(rdpcoi, Class <span class="sc">==</span> <span class="st">&quot;Actinopteri&quot;</span>)</span>
<span id="cb21-13"><a href="supplementary-workflow.html#cb21-13" tabindex="-1"></a></span>
<span id="cb21-14"><a href="supplementary-workflow.html#cb21-14" tabindex="-1"></a><span class="do">#### ------------------------- Random sampling  --------------------------------------------------------------- #### </span></span>
<span id="cb21-15"><a href="supplementary-workflow.html#cb21-15" tabindex="-1"></a></span>
<span id="cb21-16"><a href="supplementary-workflow.html#cb21-16" tabindex="-1"></a>random_sampling_list <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb21-17"><a href="supplementary-workflow.html#cb21-17" tabindex="-1"></a>i <span class="ot">=</span> <span class="dv">0</span> </span>
<span id="cb21-18"><a href="supplementary-workflow.html#cb21-18" tabindex="-1"></a></span>
<span id="cb21-19"><a href="supplementary-workflow.html#cb21-19" tabindex="-1"></a><span class="cf">for</span> (genus <span class="cf">in</span> <span class="fu">unique</span>(rdp_coi_actino<span class="sc">$</span>Genus)){</span>
<span id="cb21-20"><a href="supplementary-workflow.html#cb21-20" tabindex="-1"></a>  i <span class="ot">=</span> i <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb21-21"><a href="supplementary-workflow.html#cb21-21" tabindex="-1"></a>  sub_df <span class="ot">=</span> <span class="fu">subset</span>(rdp_coi_actino, Genus <span class="sc">==</span> genus)</span>
<span id="cb21-22"><a href="supplementary-workflow.html#cb21-22" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">nrow</span>(sub_df) <span class="sc">&gt;</span> <span class="dv">10</span>) {</span>
<span id="cb21-23"><a href="supplementary-workflow.html#cb21-23" tabindex="-1"></a>    rand_sub_df <span class="ot">=</span> sub_df[<span class="fu">sample</span>(<span class="fu">nrow</span>(sub_df), <span class="at">size=</span><span class="dv">10</span>), ] </span>
<span id="cb21-24"><a href="supplementary-workflow.html#cb21-24" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb21-25"><a href="supplementary-workflow.html#cb21-25" tabindex="-1"></a>    rand_sub_df <span class="ot">=</span> sub_df</span>
<span id="cb21-26"><a href="supplementary-workflow.html#cb21-26" tabindex="-1"></a>  }</span>
<span id="cb21-27"><a href="supplementary-workflow.html#cb21-27" tabindex="-1"></a>    random_sampling_list[[i]] <span class="ot">=</span> rand_sub_df</span>
<span id="cb21-28"><a href="supplementary-workflow.html#cb21-28" tabindex="-1"></a>}</span>
<span id="cb21-29"><a href="supplementary-workflow.html#cb21-29" tabindex="-1"></a></span>
<span id="cb21-30"><a href="supplementary-workflow.html#cb21-30" tabindex="-1"></a><span class="do">#### ------------------------- Generate list of headers to keep ----------------------------------------------- #### </span></span>
<span id="cb21-31"><a href="supplementary-workflow.html#cb21-31" tabindex="-1"></a></span>
<span id="cb21-32"><a href="supplementary-workflow.html#cb21-32" tabindex="-1"></a>df <span class="ot">=</span> <span class="fu">do.call</span>(<span class="st">&quot;rbind&quot;</span>, random_sampling_list)</span>
<span id="cb21-33"><a href="supplementary-workflow.html#cb21-33" tabindex="-1"></a></span>
<span id="cb21-34"><a href="supplementary-workflow.html#cb21-34" tabindex="-1"></a>df <span class="ot">=</span> df <span class="sc">%&gt;%</span> <span class="fu">unite</span>(<span class="st">&quot;sequence_header&quot;</span>, ID<span class="sc">:</span>Species, <span class="at">remove =</span> <span class="cn">FALSE</span>, <span class="at">sep  =</span> <span class="st">&quot;;&quot;</span>)</span>
<span id="cb21-35"><a href="supplementary-workflow.html#cb21-35" tabindex="-1"></a><span class="fu">write.table</span>(df<span class="sc">$</span>sequence_header, <span class="st">&quot;/home/kvilleneuve/fish_edna/database/taxxi_cvi/COI_DB/rdp_coi_headers_to_keep.txt&quot;</span>, <span class="at">col.names =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li>Pass the file generated by the R script to seqkit to create a subset database</li>
</ol>
<div class="sourceCode" id="cb22"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb22-1"><a href="supplementary-workflow.html#cb22-1" tabindex="-1"></a><span class="ex">seqkit</span> grep <span class="at">-nrf</span> rdp_coi_headers_to_keep.txt chord_rdp_coi.fasta <span class="at">-o</span> curated_rdp_coi.fasta</span></code></pre></div>
</div>
<div id="barque" class="section level3 unnumbered hasAnchor">
<h3>Barque<a href="supplementary-workflow.html#barque" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Remove sequence of taxa which do not belong to phylum Chordata</li>
</ol>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb23-1"><a href="supplementary-workflow.html#cb23-1" tabindex="-1"></a><span class="ex">seqkit</span> grep <span class="at">-r</span> <span class="at">-n</span> <span class="at">-p</span> <span class="st">&#39;.*chordata_*&#39;</span> barque_coi.fasta <span class="at">-o</span> chord_barque_coi.fasta</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Add unique ID to every sequence header, this is required to randomly sample 10 sequences per genus</li>
</ol>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb24-1"><a href="supplementary-workflow.html#cb24-1" tabindex="-1"></a><span class="fu">awk</span> <span class="st">&#39;/^&gt;/{sub(&quot;&gt;&quot;, &quot;&gt;&quot;++i&quot;_&quot;)}1&#39;</span> chord_barque_coi.fasta <span class="op">&gt;</span> chord_barque_coi_ID.fasta</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Extract sequences header and print into txt file</li>
</ol>
<div class="sourceCode" id="cb25"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb25-1"><a href="supplementary-workflow.html#cb25-1" tabindex="-1"></a><span class="fu">grep</span> <span class="st">&quot;&gt;&quot;</span> chord_barque_coi_ID.fasta <span class="kw">|</span> <span class="fu">sed</span> <span class="at">-e</span> <span class="st">&quot;s/&gt;//&quot;</span> <span class="op">&gt;</span> barque_coi_ID_header.txt</span>
<span id="cb25-2"><a href="supplementary-workflow.html#cb25-2" tabindex="-1"></a><span class="fu">grep</span> <span class="st">&quot;&gt;&quot;</span> rdp_coi.fasta <span class="kw">|</span> <span class="fu">sed</span> <span class="at">-e</span> <span class="st">&quot;s/&gt;//&quot;</span> <span class="op">&gt;</span> rdp_coi_header.txt</span></code></pre></div>
<ol start="4" style="list-style-type: decimal">
<li>The file with the sequence headers can then be passed to R to generate a list of which headers to keep. As per methodology by Edgar (2018), because the databases have a highly uneven numbers of sequences per genus, a subset is create by imposing a maximum of 10 sequences per genus. Headers are sampled at random to meet this constraint.</li>
</ol>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="supplementary-workflow.html#cb26-1" tabindex="-1"></a><span class="do">#### ------------------------- Load libraries ------------------------------------------------------------------####  </span></span>
<span id="cb26-2"><a href="supplementary-workflow.html#cb26-2" tabindex="-1"></a></span>
<span id="cb26-3"><a href="supplementary-workflow.html#cb26-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb26-4"><a href="supplementary-workflow.html#cb26-4" tabindex="-1"></a></span>
<span id="cb26-5"><a href="supplementary-workflow.html#cb26-5" tabindex="-1"></a><span class="do">#### ------------------------- Load databases sequence header -------------------------------------------------- #### </span></span>
<span id="cb26-6"><a href="supplementary-workflow.html#cb26-6" tabindex="-1"></a></span>
<span id="cb26-7"><a href="supplementary-workflow.html#cb26-7" tabindex="-1"></a></span>
<span id="cb26-8"><a href="supplementary-workflow.html#cb26-8" tabindex="-1"></a>barquecoi <span class="ot">=</span> <span class="fu">read.table</span>(<span class="st">&quot;/home/kvilleneuve/fish_edna/database/taxxi_cvi/COI_DB/barque_coi_ID_header.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb26-9"><a href="supplementary-workflow.html#cb26-9" tabindex="-1"></a>rdpcoi <span class="ot">=</span> <span class="fu">read.table</span>(<span class="st">&quot;/home/kvilleneuve/fish_edna/database/taxxi_cvi/COI_DB/rdp_coi_header.txt&quot;</span>,<span class="at">sep =</span> <span class="st">&quot;;&quot;</span>)</span>
<span id="cb26-10"><a href="supplementary-workflow.html#cb26-10" tabindex="-1"></a></span>
<span id="cb26-11"><a href="supplementary-workflow.html#cb26-11" tabindex="-1"></a><span class="do">#### ------------------------- Format dataframes for merging --------------------------------------------------- #### </span></span>
<span id="cb26-12"><a href="supplementary-workflow.html#cb26-12" tabindex="-1"></a></span>
<span id="cb26-13"><a href="supplementary-workflow.html#cb26-13" tabindex="-1"></a><span class="co"># The sequence headers of the COI database used by barque has the following format : Phylum_Genus_Species</span></span>
<span id="cb26-14"><a href="supplementary-workflow.html#cb26-14" tabindex="-1"></a><span class="co"># Therefore, to keep only sequence belonging to class Actinopteri </span></span>
<span id="cb26-15"><a href="supplementary-workflow.html#cb26-15" tabindex="-1"></a><span class="co"># I am merging the header from the barque COI database with headers from the RDP COI database to get complete </span></span>
<span id="cb26-16"><a href="supplementary-workflow.html#cb26-16" tabindex="-1"></a><span class="co"># taxonomic rank</span></span>
<span id="cb26-17"><a href="supplementary-workflow.html#cb26-17" tabindex="-1"></a></span>
<span id="cb26-18"><a href="supplementary-workflow.html#cb26-18" tabindex="-1"></a>barquecoi<span class="sc">$</span>Species <span class="ot">=</span> <span class="fu">paste</span>(barquecoi<span class="sc">$</span>V2, barquecoi<span class="sc">$</span>V3, <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>) <span class="co"># Create column with specie name </span></span>
<span id="cb26-19"><a href="supplementary-workflow.html#cb26-19" tabindex="-1"></a>barquecoi_unique <span class="ot">=</span> barquecoi[<span class="sc">!</span><span class="fu">duplicated</span>(barquecoi), ] <span class="co"># Keeping only unique value</span></span>
<span id="cb26-20"><a href="supplementary-workflow.html#cb26-20" tabindex="-1"></a></span>
<span id="cb26-21"><a href="supplementary-workflow.html#cb26-21" tabindex="-1"></a><span class="co"># Add column names to RDP database </span></span>
<span id="cb26-22"><a href="supplementary-workflow.html#cb26-22" tabindex="-1"></a><span class="fu">names</span>(rdpcoi) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;ID&quot;</span>, <span class="st">&quot;Cellular_organism&quot;</span>,<span class="st">&quot;Superkingdom&quot;</span>, <span class="st">&quot;Kingdom&quot;</span>, <span class="st">&quot;Phylum&quot;</span>, <span class="st">&quot;Class&quot;</span>, <span class="st">&quot;Order&quot;</span>, <span class="st">&quot;Family&quot;</span>, <span class="st">&quot;Genus&quot;</span>, <span class="st">&quot;Species&quot;</span>)</span>
<span id="cb26-23"><a href="supplementary-workflow.html#cb26-23" tabindex="-1"></a>rdpcoi_noID <span class="ot">=</span> <span class="fu">subset</span>(rdpcoi, <span class="at">select =</span> <span class="sc">-</span><span class="fu">c</span>(ID)) <span class="co"># Remove column ID (they are unique values)</span></span>
<span id="cb26-24"><a href="supplementary-workflow.html#cb26-24" tabindex="-1"></a>rdpcoi_unique <span class="ot">=</span> rdpcoi_noID[<span class="sc">!</span><span class="fu">duplicated</span>(rdpcoi_noID), ] <span class="co"># Keeping only unique value</span></span>
<span id="cb26-25"><a href="supplementary-workflow.html#cb26-25" tabindex="-1"></a></span>
<span id="cb26-26"><a href="supplementary-workflow.html#cb26-26" tabindex="-1"></a><span class="do">#### ------------------------- Merge dataframes  -------------------------------------------------------------- #### </span></span>
<span id="cb26-27"><a href="supplementary-workflow.html#cb26-27" tabindex="-1"></a></span>
<span id="cb26-28"><a href="supplementary-workflow.html#cb26-28" tabindex="-1"></a>comb_df <span class="ot">=</span> <span class="fu">merge</span>(barquecoi_unique, rdpcoi_unique, <span class="at">by =</span> <span class="st">&quot;Species&quot;</span>, <span class="at">all.y =</span> <span class="cn">FALSE</span> )</span>
<span id="cb26-29"><a href="supplementary-workflow.html#cb26-29" tabindex="-1"></a></span>
<span id="cb26-30"><a href="supplementary-workflow.html#cb26-30" tabindex="-1"></a><span class="do">#### ------------------------- Random sampling  --------------------------------------------------------------- #### </span></span>
<span id="cb26-31"><a href="supplementary-workflow.html#cb26-31" tabindex="-1"></a></span>
<span id="cb26-32"><a href="supplementary-workflow.html#cb26-32" tabindex="-1"></a>random_sampling_list <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb26-33"><a href="supplementary-workflow.html#cb26-33" tabindex="-1"></a>i <span class="ot">=</span> <span class="dv">0</span> </span>
<span id="cb26-34"><a href="supplementary-workflow.html#cb26-34" tabindex="-1"></a></span>
<span id="cb26-35"><a href="supplementary-workflow.html#cb26-35" tabindex="-1"></a><span class="cf">for</span> (genus <span class="cf">in</span> <span class="fu">unique</span>(comb_df<span class="sc">$</span>Genus.x)){</span>
<span id="cb26-36"><a href="supplementary-workflow.html#cb26-36" tabindex="-1"></a>  i <span class="ot">=</span> i <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb26-37"><a href="supplementary-workflow.html#cb26-37" tabindex="-1"></a>  sub_df <span class="ot">=</span> <span class="fu">subset</span>(comb_df, Genus.x <span class="sc">==</span> genus)</span>
<span id="cb26-38"><a href="supplementary-workflow.html#cb26-38" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">nrow</span>(sub_df) <span class="sc">&gt;</span> <span class="dv">10</span>) {</span>
<span id="cb26-39"><a href="supplementary-workflow.html#cb26-39" tabindex="-1"></a>    rand_sub_df <span class="ot">=</span> sub_df[<span class="fu">sample</span>(<span class="fu">nrow</span>(sub_df), <span class="at">size=</span><span class="dv">10</span>), ] </span>
<span id="cb26-40"><a href="supplementary-workflow.html#cb26-40" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb26-41"><a href="supplementary-workflow.html#cb26-41" tabindex="-1"></a>    rand_sub_df <span class="ot">=</span> sub_df</span>
<span id="cb26-42"><a href="supplementary-workflow.html#cb26-42" tabindex="-1"></a>  }</span>
<span id="cb26-43"><a href="supplementary-workflow.html#cb26-43" tabindex="-1"></a>    random_sampling_list[[i]] <span class="ot">=</span> rand_sub_df</span>
<span id="cb26-44"><a href="supplementary-workflow.html#cb26-44" tabindex="-1"></a>}</span>
<span id="cb26-45"><a href="supplementary-workflow.html#cb26-45" tabindex="-1"></a></span>
<span id="cb26-46"><a href="supplementary-workflow.html#cb26-46" tabindex="-1"></a>df <span class="ot">=</span> <span class="fu">do.call</span>(<span class="st">&quot;rbind&quot;</span>, random_sampling_list)</span>
<span id="cb26-47"><a href="supplementary-workflow.html#cb26-47" tabindex="-1"></a></span>
<span id="cb26-48"><a href="supplementary-workflow.html#cb26-48" tabindex="-1"></a><span class="do">#### ------------------------- Generate list of headers to keep ----------------------------------------------- #### </span></span>
<span id="cb26-49"><a href="supplementary-workflow.html#cb26-49" tabindex="-1"></a></span>
<span id="cb26-50"><a href="supplementary-workflow.html#cb26-50" tabindex="-1"></a>df<span class="sc">$</span>barque_coi_header <span class="ot">=</span> <span class="fu">paste</span>(df<span class="sc">$</span>ID, df<span class="sc">$</span>Phylum.x, df<span class="sc">$</span>Species, <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb26-51"><a href="supplementary-workflow.html#cb26-51" tabindex="-1"></a></span>
<span id="cb26-52"><a href="supplementary-workflow.html#cb26-52" tabindex="-1"></a><span class="fu">write.table</span>(df<span class="sc">$</span>barque_coi_header, <span class="st">&quot;/home/kvilleneuve/fish_edna/database/taxxi_cvi/COI_DB/barque_coi_headers_to_keep.txt&quot;</span>,<span class="at">col.names =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<ol start="5" style="list-style-type: decimal">
<li>Pass the file generated by the R script to seqkit to create a subset database.</li>
</ol>
<div class="sourceCode" id="cb27"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb27-1"><a href="supplementary-workflow.html#cb27-1" tabindex="-1"></a><span class="ex">seqkit</span> grep <span class="at">-nrf</span> barque_coi_headers_to_keep.txt chord_barque_coi.fasta <span class="at">-o</span> curated_barque_coi.fasta</span></code></pre></div>
</div>
</div>
<div id="worklow-used" class="section level2 unnumbered hasAnchor">
<h2>Worklow used<a href="supplementary-workflow.html#worklow-used" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="dada2" class="section level3 unnumbered hasAnchor">
<h3>DADA2<a href="supplementary-workflow.html#dada2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="s" class="section level4 unnumbered hasAnchor">
<h4>12S<a href="supplementary-workflow.html#s" class="anchor-section" aria-label="Anchor link to header"></a></h4>
</div>
<div id="coi" class="section level4 unnumbered hasAnchor">
<h4>COI<a href="supplementary-workflow.html#coi" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>839 ASV | from these 839 ASVs, 20 (2.38 %) are species with a bootstrap level above 70 % | from these 20 species, 8 (40 %) are from class actinopteri | from these 8 actinopteri, 7 (87.5 %) are species of freshwater fishes found in Alberta.</p>
</div>
</div>
<div id="barque-1" class="section level3 unnumbered hasAnchor">
<h3>Barque<a href="supplementary-workflow.html#barque-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>NB. Barque will only work on GNU Linux or OSX</p>
<ol style="list-style-type: decimal">
<li>Install dependencies</li>
</ol>
<ul>
<li>bash 4+</li>
<li>python 3.5+ (you can use miniconda3 to install python)</li>
<li>python distutils package</li>
<li>R 3+ (ubuntu/mint: sudo apt-get install r-base-core)</li>
<li>java (ubuntu/mint: sudo apt-get install default-jre)</li>
<li>gnu parallel
<ul>
<li>For mac OSX <code>brew install parallel</code></li>
</ul></li>
<li>flash (read merger) v1.2.11+
<ul>
<li><code>export PATH=/home/user/FLASH-1.2.11:$PATH</code></li>
</ul></li>
<li>vsearch v2.14.2+ (Barque will not work with older versions of vsearch)
<ul>
<li>Depending on how vsearch is installed either export PATH to location of vsearch or activate base conda environnement</li>
</ul></li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Download a copy of the Barque repository</li>
</ol>
<div class="sourceCode" id="cb28"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb28-1"><a href="supplementary-workflow.html#cb28-1" tabindex="-1"></a><span class="fu">git</span> clone https://github.com/enormandeau/barque</span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li><p>Edit 02_info/primers.csv to provide information describing your primers and database to use</p></li>
<li><p>Get or prepare the database(s) (see Formatting database section below) and deposit the fasta.gz file in the 03_databases folder and give it a name that matches the information of the 02_info/primers.csv file.</p></li>
</ol>
<ul>
<li><a href="https://www.ibis.ulaval.ca/services/bioinformatique/barque_databases/">Link</a> to BOLD database</li>
</ul>
<ol start="5" style="list-style-type: decimal">
<li>Modify the parameters in 02_info/barque_config.sh for your run</li>
</ol>
<ul>
<li>for COI marker :</li>
</ul>
<p><code>COI_kv,CGTATTTGGYGCYTGRGCCGGRATAGT,CARAARCTYATRTTRTTYATTCG,100,450,bold,0.97,0.9,0.85</code></p>
<ul>
<li>for 12S marker</li>
</ul>
<p><code>12s200pb,GTCGGTAAAACTCGTGCCAGC,CATAGTGGGGTATCTAATCCCAGTTTG,150,350,12S,0.98,0.9,0.85</code></p>
<ol start="6" style="list-style-type: decimal">
<li>Launch Barque</li>
</ol>
<p>Move into the Github directory and launch barque</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb29-1"><a href="supplementary-workflow.html#cb29-1" tabindex="-1"></a><span class="ex">./barque</span> 02_info/barque_config.sh</span></code></pre></div>
<p>Once the pipeline has finished running, all result files are found in the 12_results folder.
After a run, it is recommended to make a copy of this folder with some additional info</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb30-1"><a href="supplementary-workflow.html#cb30-1" tabindex="-1"></a><span class="fu">cp</span> <span class="at">-r</span> 12_results 12_results_PROJECT_NAME_DATE_SOME_ADDITIONAL_INFO</span></code></pre></div>
</div>
</div>
<div id="curating-databases-for-alberta-freshwater-fish" class="section level2 unnumbered hasAnchor">
<h2>Curating databases for Alberta freshwater fish<a href="supplementary-workflow.html#curating-databases-for-alberta-freshwater-fish" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Using <code>seqkit</code> tool to filter databases in order to keep only sequences of freshwater fishes found in Alberta</p>
<p>seqkit grep -i -f ids.txt test.fa</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb31-1"><a href="supplementary-workflow.html#cb31-1" tabindex="-1"></a><span class="ex">seqkit</span> grep <span class="at">-nrf</span> ID_barque_12S.txt barque_12S.fasta <span class="at">-o</span> alberta_barque_12S.fasta</span>
<span id="cb31-2"><a href="supplementary-workflow.html#cb31-2" tabindex="-1"></a><span class="ex">seqkit</span> grep <span class="at">-nrf</span> ID_barque_COI.txt bold_coi_for_barque_2023-09-12.fasta <span class="at">-o</span> alberta_barque_COI.fasta</span></code></pre></div>
<p>For RDP</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb32-1"><a href="supplementary-workflow.html#cb32-1" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/ /;/g&#39;</span> RDP_COI.fasta</span>
<span id="cb32-2"><a href="supplementary-workflow.html#cb32-2" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/ /;/g&#39;</span> ID_RDP_COI.txt</span>
<span id="cb32-3"><a href="supplementary-workflow.html#cb32-3" tabindex="-1"></a><span class="ex">seqkit</span> grep <span class="at">-nrf</span> ID_RDP_COI.txt RDP_COI.fasta <span class="at">-o</span> alberta_RDP_COI.fasta</span>
<span id="cb32-4"><a href="supplementary-workflow.html#cb32-4" tabindex="-1"></a></span>
<span id="cb32-5"><a href="supplementary-workflow.html#cb32-5" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/\t/;/g&#39;</span> 12S_rdp.fasta</span>
<span id="cb32-6"><a href="supplementary-workflow.html#cb32-6" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-i</span> <span class="st">&#39;s/\t/;/g&#39;</span> ID_RDP_12S.txt</span>
<span id="cb32-7"><a href="supplementary-workflow.html#cb32-7" tabindex="-1"></a><span class="ex">seqkit</span> grep <span class="at">-nrf</span> ID_RDP_12S.txt 12S_rdp.fasta <span class="at">-o</span> alberta_RDP_12S.fasta</span></code></pre></div>

</div>
</div>
<hr>
<center>
  <div class="footer">
      Karine Villeneuve
      <p : id="date"></p>
<script>
  // Get current date and time
//  var now = new Date();
//  var date = now.toLocaleString();
//  n =  new Date();
//  y = n.getFullYear();
//  m = n.getMonth();
//  d = n.getDate();
//  document.getElementById("date").innerHTML = d + "/" + m + "/" + y;
  n = new Date().toDateString()
  document.getElementById("date").innerHTML = n
  // Insert date and time into HTML
//  document.getElementById("date").innerHTML = date;
</script>
  </div>
</center>
            </section>

          </div>
        </div>
      </div>
<a href="taxonomy-cross-validation-by-identity-taxxi.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="supplementary-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/karinevilleneuve/fish_eDNA/master/03_code.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
}
});
});
</script>

</body>

</html>
